File: linux/ardium_source/presentation/08_Concurrency/concurrency.py
# Topic: Concurrency (Threading)
# Run a task in background

import threading
import time

def worker(arg):
    print(f"Worker running... ID: {arg}")
    time.sleep(1)
    print("Worker done.")

def main():
    print("Main: Starting worker")
    t = threading.Thread(target=worker, args=(1,))
    t.start()
    
    print("Main: Waiting for worker")
    t.join()
    
    print("Main: Done")

if __name__ == "__main__":
    main()

File: linux/ardium_source/presentation/08_Concurrency/concurrency.rs
// Topic: Concurrency (Threads)
// Run a task in background

use std::thread;
use std::time::Duration;

fn main() {
    println!("Main: Starting worker");
    
    let handle = thread::spawn(|| {
        println!("Worker running... ID: 1");
        thread::sleep(Duration::from_secs(1));
        println!("Worker done.");
    });
    
    println!("Main: Waiting for worker");
    handle.join().unwrap();
    
    println!("Main: Done");
}
